### PR TITLE
Add options for Etcd HTTPS communication

### LINKED ISSUE #185: ETCD https setting flags are missing.
It would be nice to have the ETCD cert key and ca flags for when etcd is using https.

### PR DESCRIPTION
**Pull request by tboerger**
In order to be able to talk to a secured Etcd cluster i have added
options for assigning SSL certificates to the Etcd client.

**Fixes #185**
I really want to have that to be able t use it within my environment.

### CODE REVIEW

**Comment by tboerger** [comment]
Hum, travis is failing. Is this really related to this pull request?

**Comment by psi-4ward** [comment]
yeehaaa secure etcd support!

**Comment by tboerger** [comment]
If somebody wants to try or use it, I have created a container including my pull request at https://github.com/dockhippie/vulcand

File: engine/etcdng/etcd.go [hunk 1]
```diff
@@ -32,14 +35,17 @@ type Options struct {
 	Box             *secret.Box
 }
 
-func New(nodes []string, etcdKey string, registry *plugin.Registry, options Options) (engine.Engine, error) {
+func New(nodes []string, etcdKey, etcdCaFile, etcdCertFile, etcdKeyFile string, registry *plugin.Registry, options Options) (engine.Engine, error) {
```
Comment (klizhentas) [inline engine/etcdng/etcd.go]: Could you please move these parameters to Options instead? this will simplify the constructor

**Comment by klizhentas** [comment]
Aside from a minor comment, looks good to me, will merge once the code review comment is applied.

File: engine/etcdng/etcd.go (hunk 1 above)
Comment (tboerger) [inline engine/etcdng/etcd.go]: Thanks for the response, I will address this as soon as possible, but maybe it will take the next weekend.

Comment (klizhentas) [inline engine/etcdng/etcd.go]: Sure!

**Comment by tboerger** [comment]
@klizhentas I have updated the pull request with your comments.
Ready to merge :)

**Comment by klizhentas** [comment]
looks good to me, thanks!

**Comment by tboerger** [comment]
Great, now i can use the master branch for my docker container again :)

On alpine linux it's an awesome container size :)

**Pull request merged**